### PR TITLE
Added profile and profile account types.

### PR DESCRIPTION
**Pull request by ghazninattarshah**
Added profile and additional types and enum

[OB-916](https://brankas.atlassian.net/browse/OB-916)

### CODE REVIEW

File: v1/types/types.gunk [hunk 1]
```diff
@@ -251,3 +251,156 @@ type Location struct {
...
+	Name string `pb:"4" json:"name"`
+
+	// The title / title displayed for a contact information.
+	Title string `pb:"5" json:"title"`
+
+	// Detailed data set related to contacts
+	Details []ContactDetail `pb:"6" json:"details"`
+
+	// Set of phone numbers associated with contact information
+	Phone map[string]bool `pb:"7" json:"phone"`
```
Comment (nkristianto) [inline v1/types/types.gunk]: why this use map[string]bool?
what represented with the bool?

File: v1/types/types.gunk [hunk 2]
```diff
@@ -251,3 +251,156 @@ type Location struct {
...
+	Work
+
+	// Other
+	Custom
+)
+
+// Directory information structure
+type ContactDetail struct {
+	// Type of contact information
+	ContactDetailType ContactDetailType `pb:"1" json:"type"`
```
Comment (nkristianto) [inline v1/types/types.gunk]: still don't understand why we use const to define contact detail

File: v1/types/types.gunk (hunk 1 above)
Comment (ghazninattarshah) [reply v1/types/types.gunk]: That's from the acb proto.

File: v1/types/types.gunk (hunk 2 above)
Comment (ghazninattarshah) [reply v1/types/types.gunk]: That's from the acb proto.

**Review by nkristianto — APPROVED** [review]

**Comment by nkristianto** [comment]
@Hekewa , need your review here

**Comment by ghazninattarshah** [comment]
@Hekewa. @nkristianto I have updated the gunk definitions for only profiles and it's accounts with basic information.

File: v1/types/types.gunk (hunk 1 above)
Comment (ghazninattarshah) [reply v1/types/types.gunk]: Removed

File: v1/types/types.gunk (hunk 2 above)
Comment (ghazninattarshah) [reply v1/types/types.gunk]: Removed

File: v1/types/types.gunk [hunk 3]
```diff
@@ -251,3 +251,50 @@ type Location struct {
...
+
+	// Account identifier
+	ProfileID string `pb:"3" json:"profile_id"`
+}
+
+// Structure of customer profile information
+type Profile struct {
+
+	// ProfileID is the unique identifier of a profile.
+	ProfileID string `pb:"1" json:"profile_id"`
```
Comment (nkristianto) [inline v1/types/types.gunk:272]: can we just make id ID?
it would be profile.ID

Comment (ghazninattarshah) [reply v1/types/types.gunk:272]: I thought of making it as ID, but referred to account proto, it was AccountID so I followed the same.

**Review by Hekewa — APPROVED** [review]

**Pull request merged**